gcc/rust/backend/rust-compile-pattern.cc

```diff
@@ -915,8 +951,26 @@ CompilePatternBindings::visit (HIR::SlicePattern &pattern)
     case TyTy::TypeKind::SLICE:
       rust_sorry_at (
 	pattern.get_locus (),
-	"SlicePattern matching against slices are not yet supported");
+	"SlicePattern matching against non-ref slices are not yet supported");
       break;
+    case TyTy::TypeKind::REF:
+      {
+	tree data_field
```

Comment:
i dont think you need these now if you just use 

slice_index_expression (match_scruitinee, expr, index, locus)